include/hdlConvertor/hdlObjects/hdlTypeDec.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include <hdlConvertor/hdlObjects/named.h>
+#include <hdlConvertor/hdlObjects/iHdlObj.h>
+
+
+namespace hdlConvertor {
+namespace hdlObjects {
+
+/*
+ * HDL AST node for module declaration
+ * (part with ports and params for Verilog, Entity for VHDL)
+ * */
+class HdlTypeDec: public WithNameAndDoc, public iHdlObj {
+public:
+	enum TypeClass {hdltc_error, hdltc_enum, hdltc_array, hdltc_struct, 
```

Comment:
That makes sense.  I learned C++ over 20 years ago when RTTI was considered "dirty" somehow.  I'm happy to remove this enumeration and the `typeClass` method.

So if someone uses the C++ interface (w/o Python), are they supposed to start type casting every node in the AST to figure out what kind of object it is?  Avoiding duplication makes sense in the parsing context, but may make it harder to use in the back-end context.

Reply:
Maybe we can use a method which returns the type enum instead of a enum type stored in property.
```cpp
class a: base {
virtual enum TypeT get_type() override {return TypeT:a;}
};
```

Reply:
I think this is exactly what I did.  In your example, `TypeT` => `TypeClass` in my code, your example `get_type` => `typeClass` in my code.  I made the method abstract in the base class.  I think it's a great idea to put these methods right in the class declaration.  I didn't notice you doing this elsewhere, so I put them in the cpp file.

Does my proposed code do something different than what you're suggesting?

Reply:
> Does my proposed code do something different than what you're suggesting?

no